tests/pkg/apimachinery/certs.go

```
@@ -0,0 +1,73 @@
+package apimachinery
+
+import (
+	"crypto/x509"
+	"github.com/golang/glog"
+	"io/ioutil"
+	"k8s.io/client-go/util/cert"
+	"os"
+)
+
+type CertContext struct {
+	Cert        []byte
+	Key         []byte
+	SigningCert []byte
+}
+
+// Setup the server cert. For example, user apiservers and admission webhooks
+// can use the cert to prove their identify to the kube-apiserver
+func SetupServerCert(namespaceName, serviceName string) *CertContext {
+	certDir, err := ioutil.TempDir("", "test-e2e-server-cert")
+	if err != nil {
+		glog.Errorf("Failed to create a temp dir for cert generation %v", err)
+	}
+	defer os.RemoveAll(certDir)
+	signingKey, err := cert.NewPrivateKey()
+	if err != nil {
+		glog.Errorf("Failed to create CA private key %v", err)
+	}
+	signingCert, err := cert.NewSelfSignedCACert(cert.Config{CommonName: "e2e-server-cert-ca"}, signingKey)
+	if err != nil {
+		glog.Errorf("Failed to create CA cert for apiserver %v", err)
+	}
+	caCertFile, err := ioutil.TempFile(certDir, "ca.crt")
+	if err != nil {
+		glog.Errorf("Failed to create a temp file for ca cert generation %v", err)
```

and all these should return a error.